Core/PersistentPixelStoring.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  PersistentPixelStoring.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Networking
+
+public struct PersistentPixelMetadata: Identifiable, Codable, Equatable {
+
+    public enum PixelType: Codable {
+        case daily
+        case count
+        case regular
+    }
+
+    public let id: UUID
+    public let eventName: String
+    public let pixelType: PixelType
+    public let additionalParameters: [String: String]
+    public let includedParameters: [Pixel.QueryParameters]
+
+    public init(eventName: String,
+                pixelType: PersistentPixelMetadata.PixelType,
+                additionalParameters: [String: String],
+                includedParameters: [Pixel.QueryParameters]) {
+        self.id = UUID()
+        self.eventName = eventName
+        self.pixelType = pixelType
+        self.additionalParameters = additionalParameters
+        self.includedParameters = includedParameters
+    }
+
+    var pixelName: String {
+        switch pixelType {
+        case .daily: return eventName + "_d"
+        case .count: return eventName + "_c"
+        case .regular: return eventName
```

Comment:
I recall we wanted to move away from _d and _c - having this hardcoded here will make it harder to do so. Can we just store name along with suffix in `eventName`? I'm not sure there is any real value of storing if given pixel was daily or count other than name handling.

Reply:
The `DailyPixel` API has been updated in the following way:
1. It now accepts a tuple of suffixes when sending a pixel
2. Any code using the `fireDailyAndCount` API currently has been updated to use the legacy suffixes
3. Finally, the `fireDailyAndCount` API was given new `_daily` and `_count` suffixes by default, so that no usage of the old suffixes gets added to the codebase going forward

The `PersistentPixel` class will now use whatever suffixes are given to it, and stores the final pixel name (including suffix) to disk, so the `PersistentPixelMetadata` struct no longer cares about which type of pixel it's dealing with.

Because of step 2 the PR has grown in size a bit, but for testing it you can remove the default value from `DailyPixel` and confirm that the app still compiles, as all uses have been accounted for.